### PR TITLE
Cheerp benchmarking improvements

### PR DESCRIPTION
**Pull request by alexp-sssup**
This set of patches improves the Cheerp benchmarking capabilities. The following tests are known to be working, but others might work as well.

```base64, copy, corrections, fannkuch, fasta, havlak, memops, primes, skinning, zzz_box2d, zzz_bullet, zzz_zlib```

### CODE REVIEW

File: tests/test_benchmark.py [hunk 1]
```diff
@@ -116,7 +116,7 @@ def __init__(self, name, cc, cxx, args=[OPTIMIZATIONS]):
   def build(self, parent, filename, args, shared_args, emcc_args, native_args, native_exec, lib_builder, has_output_parser):
     self.parent = parent
     if lib_builder:
-      native_args += lib_builder(self.name, native=True, env_init={'CC': self.cc, 'CXX': self.cxx})
+      native_args += lib_builder(self.name, native=True, env_init={'CC': self.cc, 'CXX': self.cxx, 'CXXFLAGS': "-Wno-c++11-narrowing"})
```
Comment (kripken) [inline tests/test_benchmark.py]: why is this change necessary - it's for native, no cheerp?

File: tests/test_benchmark.py [hunk 2]
```diff
@@ -985,7 +969,7 @@ def lib_builder(name, native, env_init):
       return self.get_library('bullet', [os.path.join('src', '.libs', 'libBulletDynamics.a'),
                                          os.path.join('src', '.libs', 'libBulletCollision.a'),
                                          os.path.join('src', '.libs', 'libLinearMath.a')],
-                              configure_args=['--disable-demos', '--disable-dependency-tracking'], native=native, cache_name_extra=name, env_init=env_init)
+                              configure_args=['--disable-demos', '--disable-dependency-tracking', '--host=i686-unknown-linux'], native=native, cache_name_extra=name, env_init=env_init)
```
Comment (kripken) [inline tests/test_benchmark.py]: why is this flag needed? and for which platform?

Comment (alexp-sssup) [reply tests/test_benchmark.py]: This flag is needed for 2 reasons.
1) bullet in it's configure.ac tries to do platform detection and will fail on unknown platforms
2) More in general, configure will try to compile and run a test file to check if the C compiler is sane. As Cheerp will generate a wasm file which cannot be run, configure will fail. Passing ```--host``` enables cross compile mode, which lets configure complete happily.

File: tests/test_benchmark.py (hunk 1 above)
Comment (alexp-sssup) [reply tests/test_benchmark.py]: As far as I can see the benchmarking system uses fastcomp's clang to build the native test and with that bullet won't compile without the ```-Wno-c++11-narrowing``` flag.

Here is what version of fastcomp I am using

```
$ emsdk/clang/fastcomp/build_incoming_64/bin/clang++ -v
clang version 6.0.1 (https://github.com/kripken/emscripten-fastcomp-clang.git 7feb22c20d97547e66d14e6265a5b7274f8b96a9) (https://github.com/kripken/emscripten-fastcomp.git 2a333e6b14e94de7fa32dd0a772886d028515708) (emscripten 1.38.25 : 1.38.25)
```

The fact that the flag is required is surprising to me as well. Is the bullet benchmark regularly build?

**Comment by alexp-sssup** [comment]
AUTHORS updated and CI issue fixed.

File: tests/test_benchmark.py (hunk 1 above)
Comment (kripken) [reply tests/test_benchmark.py]: Thanks - I'd guess what's going on is that part of the bullet configure script is sensitive to the local system (i.e. it uses the system compiler in some cases, ignoring CC/CXX).  The benchmarks run on my machines, so your system compiler may be older/newer. Anyhow, seems fine to add.

File: tests/test_benchmark.py (hunk 2 above)
Comment (kripken) [reply tests/test_benchmark.py]: I see, thanks. Please add a comment with that (otherwise it looks very surprising! :)

File: tests/test_benchmark.py [hunk 3]
```diff
@@ -236,32 +235,51 @@ def __init__(self, name, engine, args=[OPTIMIZATIONS], binaryen_opts=[]):
...
+    static void copyArgument(int arg, char* buf)
+    {
+        client::String* a = client::args[arg];
+        for(int i=0;i<a->get_length();i++)
+            *(buf++) = a->charCodeAt(i);
+        *buf = 0;
+    }
+};
+void webMain() {
+    int arg1Len = Arguments::getArgumentLength(0);
```
Comment (kripken) [inline tests/test_benchmark.py]: This is fine to land, but fyi I am getting rid of the dependence on commandline arguments very soon - it doesn't work or is hard to use in some cases (e.g. adding commandline arguments for the web), and sometimes I send people a build and they forget to run it with an argument. Instead I'll just hardcode argc/argv into the benchmark main file before compiling it.

Comment (alexp-sssup) [reply tests/test_benchmark.py]: No problem about that, when you migrate to hardcoding the parameters just remove this code.

File: tests/test_benchmark.py (hunk 2 above)
Comment (alexp-sssup) [reply tests/test_benchmark.py]: Comment added

**Comment by kripken** [comment]
Thanks!
